### PR TITLE
Render Frame NV12 on CPU

### PR DESCRIPTION
**Pull request by dbudniko**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x ] I agree to contribute to the project under Apache 2 License.
- [x ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

```
force_builders=Custom,Custom Win,Custom Mac
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

build_image:Custom=centos:7
buildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

Xbuild_image:Custom=ubuntu-openvino-2020.3.0:16.04
Xbuild_image:Custom Win=openvino-2020.3.0
Xbuild_image:Custom Mac=openvino-2020.3.0

test_modules:Custom=gapi
test_modules:Custom Win=gapi
test_modules:Custom Mac=gapi

buildworker:Custom=linux-1
// disabled due high memory usage: test_opencl:Custom=ON
test_opencl:Custom=OFF
test_bigdata:Custom=1
test_filter:Custom=*

```

### CODE REVIEW

File: modules/gapi/include/opencv2/gapi/render/render.hpp [hunk 1]
```diff
@@ -97,6 +97,17 @@ void GAPI_EXPORTS render(cv::Mat& y_plane,
                          const Prims& prims,
                          cv::GCompileArgs&& args = {});
 
+/** @brief The function renders on two NV12 planes passed drawing primitivies
+
+@param nv12 input Media Frame :  @ref cv::MediaFrame.
+@param prims vector of drawing primitivies
+@param args graph compile time parameters
+*/
+void GAPI_EXPORTS render(cv::MediaFrame& nv12,
```
Comment (smirnov-alexey) [inline modules/gapi/include/opencv2/gapi/render/render.hpp:106]: I would put this function after the operation

File: modules/gapi/include/opencv2/gapi/render/render.hpp [hunk 2]
```diff
@@ -97,6 +97,17 @@ void GAPI_EXPORTS render(cv::Mat& y_plane,
                          const Prims& prims,
                          cv::GCompileArgs&& args = {});
 
+/** @brief The function renders on two NV12 planes passed drawing primitivies
+
+@param nv12 input Media Frame :  @ref cv::MediaFrame.
```
Comment (smirnov-alexey) [inline modules/gapi/include/opencv2/gapi/render/render.hpp:102]: Please, be consistent in `nv12/NV12`

File: modules/gapi/src/api/render.cpp [hunk 3]
```diff
@@ -69,3 +83,9 @@ cv::gapi::wip::draw::renderNV12(const cv::GMat& y,
 {
     return cv::gapi::wip::draw::GRenderNV12::on(y, uv, prims);
 }
+
+cv::GFrame cv::gapi::wip::draw::renderFrameNV12(const cv::GFrame& nv12,
+    const cv::GArray<cv::gapi::wip::draw::Prim>& prims)
```
Comment (smirnov-alexey) [inline modules/gapi/src/api/render.cpp:88]: Indentation

File: modules/gapi/include/opencv2/gapi/render/render.hpp (hunk 2 above)
Comment (dbudniko) [reply modules/gapi/include/opencv2/gapi/render/render.hpp]: done

File: modules/gapi/src/api/render.cpp (hunk 3 above)
Comment (dbudniko) [reply modules/gapi/src/api/render.cpp]: Done

**Review by smirnov-alexey — APPROVED** [review]
Looks good, thanks. I would recommend to get one more review though

File: modules/gapi/include/opencv2/gapi/render/render.hpp [hunk 4]
```diff
@@ -97,6 +97,17 @@ void GAPI_EXPORTS render(cv::Mat& y_plane,
                          const Prims& prims,
                          cv::GCompileArgs&& args = {});
 
+/** @brief The function renders on the input media frame passed drawing primitivies
+
+@param NV12 input Media Frame :  @ref cv::MediaFrame.
+@param prims vector of drawing primitivies
+@param args graph compile time parameters
+*/
+void GAPI_EXPORTS render(cv::MediaFrame& NV12,
```
Comment (rgarnov) [inline modules/gapi/include/opencv2/gapi/render/render.hpp:106]: NV12 parameter name is misleading. With such signature and function name MediaFrame here can be of the different (non nv12) format, so this function should handle all possible formats (maybe with an assert inside on which format support is currently implemented), or have a different name

File: modules/gapi/test/render/gapi_render_tests_ocv.cpp [hunk 5]
```diff
@@ -97,6 +145,19 @@ TEST_P(RenderNV12OCVTestFTexts, AccuracyTest)
...
+TEST_P(RenderFrameNV12OCVTestFTexts, AccuracyTest)
+{
+    // G-API code //////////////////////////////////////////////////////////////
+    cv::gapi::wip::draw::Prims prims;
+    prims.emplace_back(cv::gapi::wip::draw::FText{ text, org, fh, color });
+    cv::MediaFrame nv12 = cv::MediaFrame::Create<TestMediaNV12>(y_gapi_mat, uv_gapi_mat);
+    EXPECT_NO_THROW(cv::gapi::wip::draw::render(nv12, prims,
+        cv::compile_args(cv::gapi::wip::draw::freetype_font{
+        "/usr/share/fonts/truetype/wqy/wqy-microhei.ttc"
+            })));
```
Comment (rgarnov) [inline modules/gapi/test/render/gapi_render_tests_ocv.cpp:157]: Maybe it's worth to add some check that images were changed

**Review by rgarnov — CHANGES_REQUESTED** [review]

File: modules/gapi/test/render/gapi_render_tests_ocv.cpp (hunk 5 above)
Comment (dbudniko) [reply modules/gapi/test/render/gapi_render_tests_ocv.cpp:159]: Done

File: modules/gapi/include/opencv2/gapi/render/render.hpp (hunk 4 above)
Comment (dbudniko) [reply modules/gapi/include/opencv2/gapi/render/render.hpp]: Done

File: modules/gapi/include/opencv2/gapi/render/render.hpp [hunk 6]
```diff
@@ -134,6 +153,17 @@ uv image must be 8-bit unsigned planar 2-channel image @ref CV_8UC2
...
+
+/** @brief Renders Media Frame
+
+Output media frame frame cv::MediaFrame
+
+@param m_frame input image: cv::MediaFrame @ref cv::MediaFrame
+@param prims draw primitives
+*/
+GAPI_EXPORTS GFrame renderMFrame(const GFrame& m_frame,
+                                    const GArray<Prim>& prims);
```
Comment (rgarnov) [inline modules/gapi/include/opencv2/gapi/render/render.hpp:165]: indentation

**Review by rgarnov — CHANGES_REQUESTED** [review]

File: modules/gapi/src/backends/render/grenderocv.cpp [hunk 7]
```diff
@@ -114,8 +114,83 @@ GAPI_OCV_KERNEL_ST(RenderNV12OCVImpl, cv::gapi::wip::draw::GRenderNV12, RenderOC
...
+         * Rendering on NV12 via OpenCV looks like this:
+         *
+         * y --------> 1)(NV12 -> YUV) -> yuv -> 2)draw -> yuv -> 3)split -------> out_y
+         *                  ^                                         |
+         *                  |                                         |
+         * uv --------------                                          `----------> out_uv
+         *
+         *
+         * 1) Collect yuv mat from two planes, uv plain in two times less than y plane
+         *    so, upsample uv in tow times, with bilinear interpolation
```
Comment (rgarnov) [inline modules/gapi/src/backends/render/grenderocv.cpp:152]: tow -> two

**Review by rgarnov — CHANGES_REQUESTED** [review]

File: modules/gapi/include/opencv2/gapi/render/render.hpp (hunk 6 above)
Comment (dbudniko) [reply modules/gapi/include/opencv2/gapi/render/render.hpp:165]: done

File: modules/gapi/src/backends/render/grenderocv.cpp (hunk 7 above)
Comment (dbudniko) [reply modules/gapi/src/backends/render/grenderocv.cpp]: done

**Review by rgarnov — APPROVED** [review]

**Pull request merged**